code/lib/types/src/modules/core-common.ts

```diff
@@ -330,9 +330,9 @@ export interface StorybookConfig {
   /**
    * Tells Storybook where to find stories.
    *
-   * @example `['./src/*.stories.@(j|t)sx?']`
+   * @example `['./src/*.stories.@(j|t)sx?']` or `async () => [...(await myCustomStoriesEntryBuilderFunc())]`
    */
-  stories: StoriesEntry[];
+  stories: StoriesEntry[] | ((list?: StoriesEntry[]) => Promise<StoriesEntry[]>);
```

Comment:
I honestly don't know how `list` works here. In my testing it comes back `undefined` but I added it because that is how it was documented. Also happy to remove it/update the docs if it is not a thing that works.